src/cache/file_cache.cc

```diff
@@ -27,6 +27,17 @@ String ReplaceTildeInPath(const String& path) {
   return path;
 }
 
+String HashCombine(const Immutable& source, const List<Immutable>& files) {
+  if (files.empty()) {  // we want hash to be compatible with old versions
+    return source.Hash();
+  }
+  Immutable::Rope hashes_string{source.Hash()};
+  for (auto&& file : files) {
```

Comment:
It's a habit of mine: `auto&&` is shorter to write and the range-based for-loop will use `const_iterator` anyway because of `const List<Immutable>& files`.